neqo-http3/src/connection_client.rs

```diff
@@ -2364,4 +2430,676 @@ mod tests {
         }
         assert!(recv_header && recv_data);
     }
+
+    #[allow(clippy::cognitive_complexity)]
```

Comment:
I'm resigned to having old tests grow a little so that they end up with this annotation, but not having new functions start with it.  Please try to find some way to break this down into more manageable chunks.